tests/admin_views/test_autocomplete_view.py

```diff
@@ -143,19 +177,47 @@ def setUp(self):
         self.admin_login(username='super', password='secret', login_url=reverse('admin:index'))
 
     def test_select(self):
+        from selenium.webdriver.support.ui import Select
+        q1 = Question.objects.create(question='Who am I?')
+        Question.objects.bulk_create(
+            Question(question=str(i))
+            for i in range(30)  # exceed page size
+        )
         self.selenium.get(self.live_server_url + reverse('admin:admin_views_answer_add'))
         elem = self.selenium.find_element_by_css_selector('.select2-selection')
         elem.click()
-        results = self.selenium.find_element_by_css_selector('.select2-results')
-        self.assertTrue(results.is_displayed())
-        elem = results.find_element_by_css_selector('.select2-results__option')
-        elem.click()
+        result_container = self.selenium.find_element_by_css_selector('.select2-results')
+        self.assertTrue(result_container.is_displayed())
+
+        results = result_container.find_elements_by_css_selector('.select2-results__option')
+        self.assertEqual(len(results), 21)
+
+        search = self.selenium.find_element_by_css_selector('.select2-search__field')
+
+        # dynamically load next page
+        for _ in range(len(results)):
+            search.send_keys(Keys.ARROW_DOWN)
+        results = result_container.find_elements_by_css_selector('.select2-results__option')
+        self.assertEqual(len(results), 31)
+
+        # limit results
+        search.send_keys('Who')
+        results = result_container.find_elements_by_css_selector('.select2-results__option')
+        self.assertEqual(len(results), 1)
+
+        # select result
+        search.send_keys(Keys.RETURN)
+        select = Select(self.selenium.find_element_by_id('id_question'))
+        self.assertEqual(select.first_selected_option.get_attribute('value'), str(q1.pk))
 
     def test_select_multiple(self):
-        self.selenium.get(self.live_server_url + reverse('admin:admin_views_lecture_add'))
+        self.selenium.get(self.live_server_url + reverse('admin:admin_views_question_add'))
         elem = self.selenium.find_element_by_css_selector('.select2-selection')
         elem.click()
         results = self.selenium.find_element_by_css_selector('.select2-results')
         self.assertTrue(results.is_displayed())
         elem = results.find_element_by_css_selector('.select2-results__option')
         elem.click()
```

Comment:
Is this test complete? What's the purpose of clicking on "No result found."?

Reply:
It does tests no results, which I didn't test in the test before. I will add it to the test before and test multiple select here in more detail including selecting many.